rtCommon/bidsArchive.py

```diff
@@ -0,0 +1,825 @@
+            to current working directory).
+
+        Examples:
+            >>> archive = BidsArchive('dataset')
+            >>> str(archive)
+            Root: ...t-cloud/docs/tutorials/dataset | Subjects: 1 |
+            Sessions: 0 | Runs: 1
+            >>> archive = BidsArchive('/tmp/downloads/dataset')
+            >>> str(archive)
+            Root: /tmp/downloads/dataset | Subjects: 20 |
+            Sessions: 3 | Runs: 2
+        """
+        self.rootPath = os.path.abspath(rootPath)
+        # Formatting initialization logic this way enables the creation of an
+        # empty BIDS archive that an incremntal can then be appended to
+        try:
+            self.data = BIDSLayout(rootPath)
+        except Exception as e:
+            logger.debug("Failed to open dataset at %s (%s)",
+                         self.rootPath, str(e))
+            self.data: BIDSLayout = None
+
+    def __str__(self):
+        out = str(self.data)
+        if 'BIDS Layout' in out:
+            out = out.replace('BIDS Layout', 'Root')
+
+        return out
+
+    # Enable accessing underlying BIDSLayout properties without inheritance
+    def __getattr__(self, attr):
+        originalAttr = attr
+
+        # If the attr is in the format getXyz, convert to get_xyz for forwarding
+        # to the BIDSLayout object
+        pattern = re.compile("get[A-Z][a-z]+")
+        if pattern.match(attr) is not None:
+            attr = attr.lower()
+            attr = attr[0:3] + '_' + attr[3:]
+
+        if not self.isEmpty():
+            try:
+                return getattr(self.data, attr)
+            except AttributeError:
+                raise AttributeError("{} object has no attribute {}".format(
+                    self.__class__.__name__, originalAttr))
+
+    """ Utility functions """
+    @staticmethod
+    def _stripLeadingSlash(path: str) -> str:
+        """
+        Strips a leading / from the path, if it exists. This prevents paths
+        defined relative to dataset root (/sub-01/ses-01) from being interpreted
+        as being relative to the root of the filesystem.
+
+        Args:
+            path: Path to strip leading slash from.
+
+        Examples:
+            >>> path = '/sub-01/ses-01/func/sub-01_task-test_bold.nii.gz'
+            >>> BidsArchive._stripLeadingSlash(path)
+            'sub-01/ses-01/func/sub-01_task-test_bold.nii.gz'
+            >>> path = 'sub-01/ses-01/func/sub-01_task-test_bold.nii.gz'
+            'sub-01/ses-01/func/sub-01_task-test_bold.nii.gz'
+        """
+        if len(path) >= 1 and path[0] == "/":
+            return path[1:]
+        else:
+            return path
+
+    def absPathFromRelPath(self, relPath: str) -> str:
+        """
+        Makes an absolute path from the relative path within the dataset.
+        """
+        return os.path.join(self.rootPath, self._stripLeadingSlash(relPath))
+
+    def fileExistsInArchive(self, path: str) -> bool:
```

Comment:
GW: should param be relPath? How can we confirm if it is relative and not from the root of filesystem?

Reply:
TODO: Use trygetfile